### PR TITLE
IR Transmitter and Receive  for robot and base

### PR DESCRIPTION
**Pull request by AmmarKothari**
added functionality to transmit and receive messages with IR for both robot and base station

### CODE REVIEW

File: DCMotorTest/DCMotorTest.ino [hunk 1]
```diff
@@ -14,7 +15,7 @@
 #define IN4 3
 #define ENB 22
 
-#define ENCODERL1   5
+#define ENCODERL1   8
```
Comment (flemichl) [inline DCMotorTest/DCMotorTest.ino:18]: Does this mean we have to change wiring on the protoboard?

File: DCMotorTest/DCMotorTest.ino [hunk 2]
```diff
@@ -332,3 +346,97 @@ void OpenGripper() {
...
+  Serial.println("Waiting for Turn Start Signal");
+}
+
+// monitors status to determine when to send message on serial
+void TurnMonitor() { 
+  if (currentState == 0) { // not our turn waiting for go signal
+    int resp = robot_listen(startValue);
+    if (resp == 1) {
+      currentState = 1; // start turn
+      Serial.println("Starting Turn");
```
Comment (flemichl) [inline DCMotorTest/DCMotorTest.ino:364]: We will need this to notify whoever is teleoperating the Pi. Will we have the Pi waiting to read this over serial?

**Review by flemichl — COMMENTED** [review]
Base station looks great! Thanks for doing this. I'm a little confused about how the driver will be able to initiate sending the done signal. I think we'll have to provide a way to set the currentState variable over serial from the Pi?

File: DCMotorTest/DCMotorTest.ino (hunk 1 above)
Comment (AmmarKothari) [reply DCMotorTest/DCMotorTest.ino:18]: i changed the wiring on the protoboard.  I think pins 5 and 8 are functionally equivalent.

File: DCMotorTest/DCMotorTest.ino (hunk 2 above)
Comment (AmmarKothari) [reply DCMotorTest/DCMotorTest.ino:364]: O right.  i didn't think about notifying over the pi.  Can we have it send a message over serial that will be visible to the operator?  Another option is to have it light up an LED that is in the field of vision.

**Pull request merged**